File: src/game/item/AmmoPack.java
package game.item;

import game.TextureManager;
import game.Vector;
import game.creature.Creature;
import game.creature.Player;

public class AmmoPack extends Item {
	public AmmoPack(Vector pos) {
		super(pos, TextureManager.AMMOPACK);
		color = new float[] {0.3f, 0.3f, 1};
	}
	
	@Override
	public boolean pickedUpBy(Creature c) {
		try {
			Player p = (Player) c;
			p.ammo += 5;
			return true;
		} catch (ClassCastException e) {
			return false;
		}
	}
}

File: ts/game/item/AmmoPack.ts
import { TextureManager } from '../TextureManager.js';
import { Vector } from '../Vector.js';
import { Creature } from '../creature/Creature.js';
import { Player } from '../creature/Player.js';
import { Item } from './Item.js';

export class AmmoPack extends Item {
  constructor(pos: Vector) {
    super(pos, TextureManager.AMMOPACK);
  }

  pickedUpBy(c: Creature): boolean {
    if (c instanceof Player) {
      c.ammo += 5;
      return true;
    }
    return false;
  }
}
